common/dbus/platform/asio_libdbus/dbus_client.cpp

```diff
@@ -64,10 +64,12 @@ error::Error DBusClient::InitializeConnection() {
 	dbus_error_init(&dbus_error);
 	dbus_conn_.reset(dbus_bus_get(DBUS_BUS_SYSTEM, &dbus_error));
 	if (!dbus_conn_) {
-		return MakeError(
```

Comment:
If we'd wrapped it in a `unique_ptr` to begin with, this would not have happened:

```
error::Error DBusClient::InitializeConnection() {
 	DBusError dbus_error;
 	dbus_error_init(&dbus_error);
+	unique_ptr<DBusError, decltype(&dbus_error_free)> dbus_err_ptr {&dbus_error, dbus_error_free};
 	dbus_conn_.reset(dbus_bus_get_private(DBUS_BUS_SYSTEM, &dbus_error));
 	if (!dbus_conn_) {
 		auto err = MakeError(
 			ConnectionError,
 			string("Failed to get connection to system bus: ") + dbus_error.message + "["
 				+ dbus_error.name + "]");
-		dbus_error_free(&dbus_error);
 		return err;
 	}
```

Reply:
I'll consider it. It didn't happen because I forgot, the reason was I thought the stack-allocated error only had static/stack-allocated data in it.